### PR TITLE
Sara (review wanted)

### PR DESCRIPTION
**Pull request by Aqveduktis**
Link to my deployed project:
https://aqveduktis-happy-thought.herokuapp.com/thoughts

My reflections on how this week's project turned out:
I thought the project turned out well and I think it was less abstract when you know how the frontend works with the API.

Things I'd like to have clarified or explained in more detail:

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ import express from 'express'
 import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
+import { Thought } from './model'
```
Comment (KarolCislowski) [inline server.js:5]: External Model 👍 🥇

File: server.js [hunk 2]
```diff
@@ -18,11 +19,70 @@ const app = express()
...
+    res.status(201).json(thought)
+  }
+  else {
+    res.status(404).json({message: `thought with id ${thoughtId} does not exsist` , error:"couldn't like thought"})
+  }
+  }
+  catch(err){
+    res.status(400).json({message:"not a valid id", error: err})
+  }
+})
```
Comment (KarolCislowski) [inline server.js:73]: This one should by `/:thoughtId/like`

File: server.js [hunk 3]
```diff
@@ -18,11 +19,70 @@ const app = express()
...
 
+app.get('/thoughts', async (req, res)=>{
+  try {
+    const thoughts = await Thought.find().sort({createdAt:'desc'}).limit(20)
+    res.json(thoughts)
+  }
+  catch {
+    res.status(404).json({error:"cannot find thoughts"})
+  }
+})
```
Comment (KarolCislowski) [inline server.js:44]: according to documentation `get "/"` need to return 20 thoughts,

File: server.js [hunk 4]
```diff
@@ -18,11 +19,70 @@ const app = express()
...
+    const {message} = req.body
+    const thought = new Thought({message})
+    const savedThought = await thought.save()
+    res.status(201).json(savedThought)
+  }
+  catch(err){
+    res.status(400).json({message:"could not save thought", error: err.errors})
+  }
+})
+
```
Comment (KarolCislowski) [inline server.js:58]: according to documentation `post "/"` need to create a new one.

File: server.js [hunk 5]
```diff
@@ -18,11 +19,70 @@ const app = express()
...
+app.delete('/thoughts/delete/:id', async (req, res) => {
+ const {id} = req.params
+ try {
+  await Thought.findOneAndDelete({ _id: id })
+  res.json({message:`thought with id:${id} was delted`})
+  }
+  catch(err) {
+    res.status(400).json({message: "thought could not be deleted", error: err})
+  }
```
Comment (KarolCislowski) [inline server.js:83]: Nice extra endpoint 🥇

**Comment by Aqveduktis** [comment]
live page: https://cranky-kowalevski-4b6f2e.netlify.app

**Pull request closed without merging**